test/c_api/value_test.cpp

```diff
@@ -556,6 +555,93 @@ TEST_F(CApiValueTest, GetStructFieldValue) {
     kuzu_query_result_destroy(&result);
 }
 
+TEST_F(CApiValueTest, getMapNumFields) {
+    kuzu_query_result result;
+    kuzu_flat_tuple flatTuple;
+    kuzu_state state;
+    auto connection = getConnection();
+    state = kuzu_connection_query(connection,
+        (char*)"MATCH (m:movies) WHERE m.length = 2544 RETURN m.audience", &result);
+    ASSERT_EQ(state, KuzuSuccess);
+    ASSERT_TRUE(kuzu_query_result_is_success(&result));
+    ASSERT_TRUE(kuzu_query_result_has_next(&result));
+    state = kuzu_query_result_get_next(&result, &flatTuple);
+    ASSERT_EQ(state, KuzuSuccess);
+    ASSERT_FALSE(kuzu_query_result_has_next(&result));
+    kuzu_value value;
+    ASSERT_EQ(kuzu_flat_tuple_get_value(&flatTuple, 0, &value), KuzuSuccess);
+
+    uint64_t mapFields;
+    ASSERT_EQ(kuzu_value_get_map_num_fields(&value, &mapFields), KuzuSuccess);
+    ASSERT_EQ(mapFields, 1);
+
+    kuzu_query_result_destroy(&result);
+    kuzu_value_destroy(&value);
+    kuzu_flat_tuple_destroy(&flatTuple);
+}
+
+TEST_F(CApiValueTest, getMapName) {
+    kuzu_query_result result;
+    kuzu_flat_tuple flatTuple;
+    kuzu_state state;
+    auto connection = getConnection();
+    state = kuzu_connection_query(connection,
+        (char*)"MATCH (m:movies) WHERE m.length = 2544 RETURN m.audience", &result);
+    ASSERT_EQ(state, KuzuSuccess);
+    ASSERT_TRUE(kuzu_query_result_is_success(&result));
+    ASSERT_TRUE(kuzu_query_result_has_next(&result));
+    state = kuzu_query_result_get_next(&result, &flatTuple);
+    ASSERT_EQ(state, KuzuSuccess);
+    ASSERT_FALSE(kuzu_query_result_has_next(&result));
+    kuzu_value value;
+    ASSERT_EQ(kuzu_flat_tuple_get_value(&flatTuple, 0, &value), KuzuSuccess);
+
+    char* mapName;
```

Comment:
There is probably a memory leak here. The `mapName` is not destroyed.

Reply:
Should be destroyed below, line 602